econml/sklearn_extensions/ensemble.py

```diff
@@ -590,10 +586,9 @@ def _mean_fn(self, X, fn, acc, slice=None):
             n_estimators = slice[1] - slice[0]
 
         # Assign chunk of trees to jobs
-        n_jobs, _, _ = _partition_estimators(n_estimators, self.n_jobs)
+        n_jobs = min(effective_n_jobs(self.n_jobs), n_estimators)
         lock = threading.Lock()
-        Parallel(n_jobs=n_jobs, verbose=self.verbose,
-                 **_joblib_parallel_args(require="sharedmem"))(
+        Parallel(n_jobs=n_jobs, verbose=self.verbose, require="sharedmem")(
```

Comment:
I believe joblib args is doing sth more than just setting require=sharedmem. I believe it also sets backend='threading' when require='sharedmem' is set. Can we verify that the same functionality is eanbled.

Reply:
this is handled in the joblib versions we require

Reply:
There is some extra logic in the sklearn private method, but it's only applicable to versions of joblib<=0.11 while we require joblib 0.13 (which does not need their fix)